pkg/driver/node/mounter/pod_mounter_test.go

```diff
@@ -0,0 +1,379 @@
+				mountSock := mppod.PathOnHost(mpPod.podPath, mppod.KnownPathMountSock)
+				_, err := os.Create(mountSock)
+				assert.NoError(t, err)
+			}()
+
+			err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+				VolumeID: testCtx.volumeID,
+				PodID:    testCtx.podUID,
+			}, mountpoint.ParseArgs(nil))
+			if err == nil {
+				t.Errorf("mount shouldn't succeeded if Mountpoint does not receive the mount options")
+			}
+
+			ok, err := testCtx.mount.IsMountPoint(testCtx.targetPath)
+			assert.NoError(t, err)
+			if ok {
+				t.Errorf("it should unmount the target path if Mountpoint does not receive the mount options")
+			}
+		})
+
+		t.Run("Unmounts target if Mountpoint Pod fails to start", func(t *testing.T) {
+			testCtx := setup(t)
+
+			testCtx.mountSyscall = func(target string, args mountpoint.Args) (fd int, err error) {
+				// Does not do real mounting
+				return int(mountertest.OpenDevNull(t).Fd()), nil
+			}
+
+			go func() {
+				mpPod := createMountpointPod(testCtx)
+				mpPod.run()
+				mpPod.receiveMountOptions(testCtx.ctx)
+
+				// Emulate that Mountpoint failed to mount
+				mountErrorPath := mppod.PathOnHost(mpPod.podPath, mppod.KnownPathMountError)
+				err := os.WriteFile(mountErrorPath, []byte("mount failed"), 0777)
+				assert.NoError(t, err)
+			}()
+
+			err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+				VolumeID: testCtx.volumeID,
+				PodID:    testCtx.podUID,
+			}, mountpoint.ParseArgs(nil))
+			if err == nil {
+				t.Errorf("mount shouldn't succeeded if Mountpoint fails to start")
+			}
+
+			ok, err := testCtx.mount.IsMountPoint(testCtx.targetPath)
+			assert.NoError(t, err)
+			if ok {
+				t.Errorf("it should unmount the target path if Mountpoint fails to start")
+			}
+		})
+	})
+
+	t.Run("Checking if its mount point", func(t *testing.T) {
+		testCtx := setup(t)
+
+		ok, _ := testCtx.podMounter.IsMountPoint(testCtx.targetPath)
+		assert.Equals(t, false, ok)
+
+		go func() {
+			mpPod := createMountpointPod(testCtx)
+			mpPod.run()
+			mpPod.receiveMountOptions(testCtx.ctx)
+		}()
+
+		err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+			VolumeID: testCtx.volumeID,
+			PodID:    testCtx.podUID,
+		}, mountpoint.ParseArgs(nil))
+		assert.NoError(t, err)
+
+		ok, err = testCtx.podMounter.IsMountPoint(testCtx.targetPath)
+		assert.NoError(t, err)
+		assert.Equals(t, true, ok)
+	})
+
+	t.Run("Unmounting", func(t *testing.T) {
+		testCtx := setup(t)
+
+		go func() {
+			mpPod := createMountpointPod(testCtx)
+			mpPod.run()
+			mpPod.receiveMountOptions(testCtx.ctx)
+		}()
+
+		err := testCtx.podMounter.Mount(testCtx.ctx, testCtx.bucketName, testCtx.targetPath, credentialprovider.ProvideContext{
+			VolumeID: testCtx.volumeID,
+			PodID:    testCtx.podUID,
+		}, mountpoint.ParseArgs(nil))
```

Comment:
Didn't we say that we don't mount if we pass no arguments?

Reply:
No. Since `bucketName` and `target`/`fd` are seperate arguments, you can have empty Mountpoint arguments.